### ISSUE TITLE
Services from .pgservice.conf don't work in the new version

### ISSUE THREAD

**Issue by Qu4tro** [dbcli/pgcli#1034]
## Description

Using services don't work (but they did in the previous release):
`PGSERVICE=spacex pgcli` or `pgcli service=spacex`
By not working I mean: It tries to get a connection with my user account and this is the error message that I get: `FATAL:  database "me" does not exist`

Might be helpful to know the PR that caused the issue: #978 

I also, kind of rely on pgcli (thanks for this gem), so I came up with a quick patch that might be helpful for anyone having the same issue:

```
From 49582ccb7c4ec98aa9736c02d300096952d065fe Mon Sep 17 00:00:00 2001
From: Xavier Francisco <xavier@yarilabs.com>
Date: Mon, 8 Apr 2019 19:46:29 +0100
Subject: [PATCH] Fix usage of pgservice

---
 pgcli/main.py      | 9 +++++----
 pgcli/pgexecute.py | 2 +-
 2 files changed, 6 insertions(+), 5 deletions(-)

diff --git a/pgcli/main.py b/pgcli/main.py
index d7853e47..adf42283 100644
--- a/pgcli/main.py
+++ b/pgcli/main.py
@@ -398,11 +398,12 @@ class PGCli(object):
                 dsn='', **kwargs):
         # Connect to the database.
 
-        if not user:
-            user = getuser()
+        if not dsn:
+            if not user:
+                user = getuser()
 
-        if not database:
-            database = user
+            if not database:
+                database = user
 
         # If password prompt is not forced but no password is provided, try
         # getting it from environment variable.
diff --git a/pgcli/pgexecute.py b/pgcli/pgexecute.py
index 8c3bfa82..6d94ccc6 100644
--- a/pgcli/pgexecute.py
+++ b/pgcli/pgexecute.py
@@ -233,7 +233,7 @@ class PGExecute(object):
         }
         new_params.update(kwargs)
         conn_params.update({
-            k: unicode2utf8(v) for k, v in new_params.items() if v is not None
+            k: unicode2utf8(v) for k, v in new_params.items() if v
         })
 
         if 'password' in conn_params and 'dsn' in conn_params:
-- 
2.21.0
```

First part removes defaults, if dsn is set. Second part cleans the dict properly (as empty attributes are mostly empty strings and not None objects).

I've opened an issue instead of a PR, since I'm not sure of the consequences these changes would have for other use cases. "Works for me" would be a good title for the patch.

**Comment by j-bennet** [comment]
@Qu4tro Thank you for the bug report! I would suggest you open a PR anyway, because that gives us a base to start with. We also have a ton of tests and hopefully they might catch it if something is broken (although sadly they didn't in https://github.com/dbcli/pgcli/pull/978). Perhaps we need to add an integration test specifically for `service=xxx`.

**Comment by Qu4tro** [comment]
Was fixed in #1051 . 

Thanks everyone.
